integration/spark2/src/main/scala/org/apache/spark/util/AlterTableUtil.scala

```diff
@@ -172,6 +172,32 @@ object AlterTableUtil {
     }
   }
 
+  /**
+   * update schema when LONG_STRING_COLUMNS are changed
+   */
+  private def updateSchemaForLongStringColumns(
+      thriftTable: TableInfo,
+      longStringColumns: String,
+      schemaConverter: SchemaConverter
+  ) = {
+    val columns = thriftTable.getFact_table.getTable_columns.asScala
```

Comment:
Long string columns will be moved after dimension columns and before complex columns internally, it better resort here.

Reply:
yes, I was working on it